types/prosemirror-commands/index.d.ts

```diff
@@ -4,13 +4,26 @@
 //                 David Hahn <https://github.com/davidka>
 //                 Tim Baumann <https://github.com/timjb>
 //                 Patrick Simmelbauer <https://github.com/patsimm>
+//                 Mike Morearty <https://github.com/mmorearty>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 3.0
 
 import { MarkType, Node as ProsemirrorNode, NodeType, Schema } from 'prosemirror-model';
 import { EditorState, Transaction } from 'prosemirror-state';
 import { EditorView } from 'prosemirror-view';
 
+export interface Command<S extends Schema = any> {
```

Comment:
Are there any reasons for using `any` instead of `unknown` here? If not, could you please replace `any` with `unknown`?

Reply:
`unknown` is not compatible with the constraint type `Schema`. If I try `<S extends Schema = unknown>`, the compiler objects, "Type 'unknown' does not satisfy the constraint 'Schema'. (2344)"

I think `<S extends Schema = Schema>` is worth considering. `Schema` itself is defined as `Schema<N extends string = any, M extends string = any>`.

I'm a little hesitant to do that in this PR, though, because throughout the rest of all of the prosemirror packages, there is currently a consistent pattern of using `<S extends Schema = any>`.

I did a little experimentation, and I do quite like `<S extends Schema = Schema>`. I'm not quite sure in practice where it is useful, but it does feel more correct. For example, `EditorState<S extends Schema>` exposes a property `schema: S`. As things are written today, since the declaration says `EditorState<S extends Schema = any>`, my editor does not provide autocomplete in this scenario:

```ts
declare const editorState: EditorState;
editorState.schema.    // <-- I typed "." but I get no hints, since `schema` is of type `any`
```

If I change the declaration to `EditorState<S extend Schema = Schema>`, I do get autocomplete hints:

<img width="587" alt="Screen Shot 2020-07-02 at 9 29 34 AM" src="https://user-images.githubusercontent.com/207671/86386511-b3e3ba00-bc46-11ea-812f-7f245df7d5a1.png">

I would suggest leaving this line the way I have it in this PR, and then doing a separate PR to change to `<S extends Schema = Schema>` throughout all of the ProseMirror packages.

The same could be done for any other types that use `= any`. For example, `Schema` itself could be changed to `Schema<N extends string = string, M extends string = string>`.

Reply:
I'm okay with using `s extends Schema = any` for consistency reasons. Let's experiment with `s extends Schema = Schema` in another change!